### ISSUE TITLE
Add 'Detect AI Text' workflow tool (desklib/ai-text-detector-v1.01)

### ISSUE THREAD

**Issue by dtubb** [dtubb/fichero#753]
## Summary

Add a workflow tool that runs [desklib/ai-text-detector-v1.01](https://huggingface.co/desklib/ai-text-detector-v1.01) on document text and saves the result (probability + verdict) as an artifact. Lives in a new "Analysis" palette group — first member of a category for *judging* text rather than *transforming/extracting* it.

**Depends on #752** (Local Models settings tab) — model is downloaded/managed from there.

## Model

- Base: `microsoft/deberta-v3-large` (~435M params)
- Fine-tuned classifier head, single sigmoid logit
- Max input: 768 tokens (~500 words)
- Weights: ~1.7 GB fp32 (~870 MB fp16)
- Inference: PyTorch, runs on Apple Silicon via MPS backend (~1–3 s per chunk on M-series)
- Top of RAID benchmark at submission time

## Architecture

**Tool:** `fichero-engine/src/fichero/workflows/tools/ai_detection.py`

```
Inputs:  text (str) — from aggregate node, like extractors
Outputs: artifact (markdown report) + metadata
```

**Run granularity:** per-page parallel — same pattern as existing per-page extractors. Aggregate text splits on `\n\n---\n\n`, each chunk scored in parallel, aggregate score = mean of per-chunk scores weighted by chunk length.

**Artifact shape** (dual-write pattern from MEMORY.md):
- **Markdown artifact**: human-readable report with verdict badge + per-page table
- **Metadata block**:
  ```json
  {
    \"ai_probability\": 0.87,
    \"label\": \"ai\",
    \"per_page\": [{\"page\": 1, \"score\": 0.92}, ...],
    \"threshold\": 0.5,
    \"model\": \"desklib/ai-text-detector-v1.01\"
  }
  ```

**Palette:** new \"Analysis\" group (first member). Tool name \"Detect AI Text\".

## Implementation notes

- **Lazy imports**: `torch` / `transformers` imported inside the tool function, not at engine startup. Engine launch must not pay torch's ~300 MB import cost when no Analysis tools are used.
- **Drop padding**: reference code uses `padding='max_length'` which wastes compute on short pages. Use `padding=True`.
- **MPS auto-detect**: `torch.backends.mps.is_available()` → device, fall back to CPU. No user setting.
- **Threshold**: ship at 0.5 (model default). Revisit if classification feels off. No user setting in v1.
- **Custom model class**: requires `DesklibAIDetectionModel(PreTrainedModel)` wrapper from the model card (classifier head over mean-pooled DeBERTa). Define inline in the tool module.
- **Apple Intelligence cannot do this** — it's a generative model, not an encoder-classifier. fm-bridge route is not reusable; this genuinely needs the torch path.

## Tool guard

Before running, check that the model is downloaded (via #752 infrastructure). If not, fail fast with an actionable error pointing at Settings → Models. Do **not** auto-download mid-workflow.

## Out of scope

- Per-domain threshold tuning (RAID paper suggests 0.5 is calibrated only on English literary text — punt)
- Confidence intervals / uncertainty estimates
- Other classifiers (sentiment, plagiarism, language ID) — separate issues, but they'll reuse this Analysis group
- Non-English detection (model is English-only)

**Comment by dtubb** [comment]
Merged to main via ms/kg-hermeneutics (build-fixed) — 3544 tests pass, Xcode build green.
